openquake/commonlib/calc.py

```diff
@@ -631,20 +631,55 @@ class RuptureGetter(object):
     :param slice_: a slice of ruptures (default: all)
     :param grp_id: the group ID of the ruptures (default: all)
     """
-    def __init__(self, dstore, slice_=slice(None), grp_id=None):
+    @classmethod
+    def from_(cls, dstore):
+        """
+        :returns: a dictionary grp_id -> RuptureGetter instance
+        """
+        array = dstore['ruptures'].value
+        grp_ids = numpy.unique(array['grp_id'])
+        return {grp_id: cls(dstore, array['grp_id'] == grp_id, grp_id)
+                for grp_id in grp_ids}
+
+    def __init__(self, dstore, mask=None, grp_id=None):
         self.dstore = dstore
-        self.slice = slice_
+        self.mask = slice(None) if mask is None else mask
```

Comment:
Please add a comment

Reply:
I have explained the `.mask` argument in the docstring